### PR TITLE
Bool builder fix

### PR DESCRIPTION
**Pull request by philass**
Addresses [!1309](https://github.com/onnx/onnx-mlir/issues/1309)

### CODE REVIEW

**Comment by philass** [comment]
Sorry for not getting to this sooner!

I would appreciate any info on a good place to put tests. I wasn't exactly sure where they should live.

**Comment by philass** [comment]
Hey @tungld,

Could I get a review?

I think the CI is failing because I don't have the correct authorization

**Comment by tungld** [comment]
@philass thank you for the patch. It looks good. Could you fix the clang-format please?
```
--- ./src/Dialect/ONNX/ONNXOpsHelper.hpp	(original)
+++ ./src/Dialect/ONNX/ONNXOpsHelper.hpp	(reformatted)
@@ -[10](https://github.com/onnx/onnx-mlir/runs/6072722882?check_suite_focus=true#step:4:10)3,7 +103,8 @@
     mlir::PatternRewriter &rewriter, mlir::Location loc);
 
 bool isFromNone(mlir::Value value);
-mlir::Type getBroadcastedRankedType(mlir::Type type1, mlir::Type type2, mlir::Type elementType = nullptr);
+mlir::Type getBroadcastedRankedType(
+    mlir::Type type1, mlir::Type type2, mlir::Type elementType = nullptr);
```

**Comment by tungld** [comment]
@jenkins-droid test this please

**Comment by philass** [comment]
@tungld, Thanks for pointing that out.

updated with the format fix

File: utils/gen_onnx_mlir.py [hunk 1]
```diff
@@ -399,6 +399,11 @@
     'Sub',
     'Xor',
 ]
+broadcast_to_bool_list = [
```
Comment (tungld) [inline utils/gen_onnx_mlir.py:402]: This list and `custom_builder_broadcast_ops_list` are overlapped, which is confusing to users.

Could we do the same as Line 408 for `custom_builder_ops_list`, e.g.
- rename `custom_builder_broadcast_ops_list` to `custom_builder_broadcast_to_same_type_ops_list`
- rename `broadcast_to_bool_list` to `custom_builder_broadcast_to_bool_ops_list`
- then union them: `custom_builder_broadcast_ops_list = custom_builder_broadcast_to_same_type_ops_list +  custom_builder_broadcast_to_bool_ops_list`

**Comment by tungld** [comment]
@jenkins-droid test this please

File: utils/gen_onnx_mlir.py (hunk 1 above)
Comment (philass) [reply utils/gen_onnx_mlir.py:402]: Yeah sure! That makes sense. I'll get right on that

**Comment by tungld** [comment]
@jenkins-droid test this please

**Review by tungld — APPROVED** [review]
LGTM! Thanks for the fix!

Please fix DCO so that we can merge the patch. Thanks!

**Comment by philass** [comment]
Updated the DCO,

Thanks @tungld for taking the time to review and provide thoughtful feedback on the MR!

Hopefully this is the first of many contributions :)

**Comment by tungld** [comment]
@jenkins-droid test this please

**Comment by tungld** [comment]
> Hopefully this is the first of many contributions :)

I see. Congratulations on the first contribution in onnx-mlir! Looking forward to your next contribution.

**Pull request merged**